### PR TITLE
Add schema name as Executor argument & fix default schema

### PR DESCRIPTION
**Pull request by Vincz**
- Add schemaName in the ExecutorArgumentsEvent
- Make the GraphQL collector retrieve schema from ExecutorArgument instead of routing
- Consider default schema as the one with name 'default' or the first one defined
- Apply same logic to Annotation Parser
- Update doc

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | yes
| Fixed tickets | #608, #712
| License       | MIT

### CODE REVIEW

File: src/Config/Parser/AnnotationParser.php [hunk 1]
```diff
@@ -253,19 +253,20 @@ private static function typeAnnotationToGQLConfiguration(
...
         if (isset($configs['definitions']['schema'])) {
+            $defaultSchemaName = isset($configs['definitions']['schema']['default']) ? 'default' : array_key_first($configs['definitions']['schema']);
             foreach ($configs['definitions']['schema'] as $schemaName => $schema) {
                 $schemaQuery = $schema['query'] ?? null;
                 $schemaMutation = $schema['mutation'] ?? null;
 
                 if ($schemaQuery && $gqlName === $schemaQuery) {
                     $isRoot = true;
-                    if ('default' == $schemaName) {
+                    if ($defaultSchemaName == $schemaName) {
```
Comment (mcg-web) [inline src/Config/Parser/AnnotationParser.php:263]: can we use strict check here?

File: src/DataCollector/GraphQLCollector.php [hunk 2]
```diff
@@ -61,8 +65,10 @@ public function getCount(): int
 
     /**
      * Return the targeted schema.
+     *
+     * @return string|false
      */
-    public function getSchema(): string
+    public function getSchema()
```
Comment (mcg-web) [inline src/DataCollector/GraphQLCollector.php:71]: this will always return string

File: src/Config/Parser/AnnotationParser.php [hunk 3]
```diff
@@ -253,19 +253,20 @@ private static function typeAnnotationToGQLConfiguration(
...
                         $isDefault = true;
                     }
                 } elseif ($schemaMutation && $gqlName === $schemaMutation) {
                     $isMutation = true;
                     $isRoot = true;
-                    if ('default' == $schemaName) {
+                    if ($defaultSchemaName == $schemaName) {
```
Comment (mcg-web) [inline src/Config/Parser/AnnotationParser.php:269]: can we use strict check here?

File: src/Config/Parser/AnnotationParser.php (hunk 1 above)
Comment (murtukov) [reply src/Config/Parser/AnnotationParser.php:263]: And a think the "if" conditions could also be simplified with `$schemaQuery === $gqlName` and `$schemaMutation === $gqlName`

File: src/DataCollector/GraphQLCollector.php [hunk 4]
```diff
@@ -28,15 +28,19 @@ public function collect(Request $request, Response $response, Throwable $excepti
     {
         $error = false;
         $count = 0;
+        $schema = false;
```
Comment (mcg-web) [inline src/DataCollector/GraphQLCollector.php:31]: use null intead

File: src/DataCollector/GraphQLCollector.php (hunk 2 above)
Comment (mcg-web) [inline src/DataCollector/GraphQLCollector.php:71]: `: ?string`

**Pull request merged**